frontend/lib/screens/chat_screen/chat_screen.dart

```diff
@@ -0,0 +1,227 @@
+                          itemCount: widget._messages.length,
+                          reverse: true,
+                          itemBuilder: (_, index) {
+                            final message = widget._messages[index];
+                            final messageSender = message.sender;
+                            return MessageCard(
+                              message: message,
+                              isByMe: (messageSender == Sender.STUDENT &&
+                                      loggedUserType == AccountType.Student) ||
+                                  (messageSender == Sender.EMPLOYER &&
+                                      loggedUserType == AccountType.Employer),
+                            );
+                          }),
+                    ),
+                    Row(
+                      mainAxisAlignment: MainAxisAlignment.center,
+                      crossAxisAlignment: CrossAxisAlignment.center,
+                      mainAxisSize: MainAxisSize.max,
+                      children: [
+                        Expanded(
+                          child: _sendingMessage
+                              ? Center(
+                                  child: CircularProgressIndicator(),
+                                )
+                              : !_creatingNewMessage && !_sendingMessage
+                                  ? Padding(
+                                      padding: const EdgeInsets.symmetric(
+                                          vertical: 16.0),
+                                      child: RaisedButton(
+                                        color: Colors.grey[600],
+                                        onPressed:
+                                            _createNewMessageButtonPressed,
+                                        child: Text(
+                                          'Create new message',
+                                          style: TextStyle(color: Colors.white),
+                                        ),
+                                      ),
+                                    )
+                                  : _createSendMessageButton(screenSize),
+                        ),
+                      ],
+                    )
+                  ],
+                )
+              : Center(
+                  child: CircularProgressIndicator(),
+                ),
+        ),
+      ),
+    );
+  }
+
+  _createNewMessageButtonPressed() {
+    setState(() {
+      _creatingNewMessage = true;
+    });
+  }
+
+  _sendMessageButtonPressed() {
+    if (_formKey.currentState.validate()) {
+      setState(() {
+        _sendingMessage = true;
+      });
+      APIService.route(
+        ENDPOINTS.Messages,
+        "/message",
+        body: Message(
+          studentId: widget.args.studentId,
+          employerId: widget.args.employerId,
+          message: _newMessage,
+          sender: AuthService().loggedUser.type == AccountType.Student
```

Comment:
The sender is always saved as a STUDENT, also when it should be an EMPLOYER

Reply:
This happens because it takes the value from the logged user and right now that is a fixed value on the AuthService